.github/workflows/reusable-macos.yml

```diff
@@ -47,11 +47,23 @@ jobs:
           ${{ inputs.free-threading && '--disable-gil' || '' }} \
           --prefix=/opt/python-dev \
           --with-openssl="$(brew --prefix openssl@3.0)"
+    - name: Build CPython for compiler warning check
+      if : ${{ !inputs.free-threading && inputs.os == 'macos-13 }}
+      run: set -o pipefail; gmake -j8 --output-sync 2>&1 | tee compiler_output_macos.txt
```

Comment:
Nit: let's move this step after the "Build CPython" step.

It moves it a little closer to "Check compiler warnings", so the two `if: ${{ !inputs.free-threading }} ...` ones are nearer each other and don't have the `if: ${{ inputs.free-threading }} ...` in between.

Reply:
Rearranged step order